Imitation Dynamics and Regret Minimization models added to the repo (#229)
drvinceknight/Nashpy: docs/how-to/solve-with-regret-minimization.rst

.. _how-to-use-regret-minimization:

Solve with Regret Minimization
==============================

One of the algorithms implemented in :code:`Nashpy` is called
:code:`regret_minimization()`, this is implemented as a method on the :code:`Game`
class::

    >>> import nashpy as nash
    >>> import numpy as np
    >>> A = np.array([[3, -1], [-1, 3]])
    >>> B = np.array([[-3, 1], [1, -3]])
    >>> rps = nash.Game(A,B)

> This :code:`regret_minimization` method returns a generator of all the
> equilibria::

Reviewer: ```suggestion
This :code:`regret_minimization` method returns a a generator of the outcomes 
of the regret minimization algorithm::
```

.. _how-to-use-regret-minimization:

> Solve with support enumeration

Reviewer: ```suggestion
Solve with Regret Minimization
```

Author: Gosh, I don't know how I missed it; copy-paste an error—changed in the latest commit.

==============================

One of the algorithms implemented in :code:`Nashpy` is called
:code:`regret_minimization()`, this is implemented as a method on the :code:`Game`
class::

    >>> import nashpy as nash
    >>> import numpy as np
    >>> A = np.array([[3, -1], [-1, 3]])
    >>> B = np.array([[-3, 1], [1, -3]])
    >>> rps = nash.Game(A,B)

This :code:`regret_minimization` method returns a generator of all the
equilibria::

    >>> ne_regret_mini = rps.regret_minimization()
    >>> print(list(ne_regret_mini))
    [([0.5, 0.5], [0.5, 0.5])]

> The program will generate more controlled results by adding the two input parameters :code:`learning_rate` and :code:`max_iterations` within the function :code:`regret_minimization`.

Reviewer: ```suggestion
:code:`regret_minimization` takes the following parameters :code:`learning_rate` and :code:`iterations`.
```

Author: Changed in the latest commit.

The Example call statements to run is given below::

    >>> A = np.array([[3, -1,3], [-1, 3,6], [-1, 1,2]])
    >>> B = np.array([[-3, 1,4], [1, -3,3], [-1, 3,4]])
    >>> rps = nash.Game(A,B)
    >>> learning_rate = 0.2
    >>> max_iterations = 1000
    >>> ne_regret_mini = rps.regret_minimization(learning_rate=learning_rate,

>     max_iterations=max_iterations)

Reviewer: ```suggestion
    iterations=iterations)
```
